PHP 8.1 Compatibility Fixes

kirmorozov opened the pull request:
PR fixes build process for PHP. 8.1,
Breaking changes from PHP source:
removed MYSQLND_SZ_T_SPEC 
https://github.com/php/php-src/commit/b80767e219dbcda5412a5d910c6430b83b747c6e
Change of mysqlnd_error_info_init 
https://github.com/php/php-src/commit/7088cffe8a2643a28aa569c850f247b8e3c0b1e4

kirmorozov committed:
PHP 8.1 Compatibility Fix mysqlnd_error_info_init

Source: https://github.com/php/php-src/commit/7088cffe8a2643a28aa569c850f247b8e3c0b1e4

kirmorozov committed:
PHP 8.1 Compatibility Fix MYSQLND_SZ_T_SPEC

Source: https://github.com/php/php-src/commit/b80767e219dbcda5412a5d910c6430b83b747c6e

temuri416 commented:
Thanks for the fix, @kirmorozov. Any idea when this going to be merged?

temuri416 commented:
btw there's a syntax error - double closing parenthesis:

https://github.com/php/pecl-database-mysql_xdevapi/pull/9/commits/11fc8630f9e11ccf7f33f560382fe3e5520fe430#diff-7561e12b39e0462dcc1c6c7ddcd6ff63a7f63e82e197a0a718a05fe0407a4f4cR779

kirmorozov committed:
Fixed typo

Change of mysqlnd_error_info_init
https://github.com/php/php-src/commit/7088cffe8a2643a28aa569c850f247b8e3c0b1e4

cmb69 commented:
I can confirm that these changes are required for PHP 8.1 compatibility.

rsomla1 commented:
Thank you for the fix @kirmorozov. We will include it in the next release. Sorry it took so long...

temuri416 commented:
@rsomla1 Any ETA on the patch? Thank you!

rsomla1 commented:
Hi, It will be part of 8.0.30 release due in a month or so.

SugarD-x commented:
Sorry to be a pest, but any news on this patch and/or that version being released? It's been quite awhile and neither has come to light yet.

rsomla1 commented:
Hi, it was included in 8.0.30 and is part of `release/8.0.30` branch: https://github.com/php/pecl-database-mysql_xdevapi/commit/e7b03c16795c56644721e719d66432cbfd9ec966

But, for some reason, it is not in `trunk`. I will check what is going on there.
